src/backend/distributed/utils/node_metadata.c

```diff
@@ -364,12 +364,21 @@ RemoveNodeFromCluster(char *nodeName, int32 nodePort, bool forceRemove)
 	bool hasShardPlacements = false;
 	WorkerNode *workerNode = NULL;
 	List *referenceTableList = NIL;
+	uint32 deletedNodeId = 0;
 
 	EnsureSchemaNode();
 	EnsureSuperUser();
 
 	workerNode = FindWorkerNode(nodeName, nodePort);
 
+	if (workerNode == NULL)
```

Comment:
I am not convinced for the necessity of this, but in order to reduce the number of modified places, I have made the change.